MOxUnit/@MOxUnitTestResult/report_xml.m

```diff
@@ -0,0 +1,164 @@
+    classpath = fullfile(p,n);
+    % Remove the leading slash
+    if strcmp(classpath(1), '/')
+        classpath = classpath(2:end);
+    end
+    % Swap / separators for . separators, to resemble object oriented
+    % languages which JUnit is initially for.
+    %classpath = strrep(classpath, filesep, '.');
+
+    % Inject the parameters into the template
+    x = sprintf(x, classpath, get(t,'location'), get(t,'name'), dur);
+
+    switch lower(c(1))
+        case 'e'
+            % Fill the contents for an error
+            es = reason.stack(1);
+            reason.message = strip_xml(reason.message);
+            x = [x, ' line="' num2str(es.line) '"' ...
+                    '>' ...
+                    '<system-err message="' ...
+                        get_first_line(reason.message) ...
+                        '">' ...
+                    reason.message ...
+                    '</system-err>' ...
+                    '</testcase>' ...
+                    ];
+
+        case 'f'
+            % Fill the contents for a failure
+            es = reason.stack(1);
+            reason.message = strip_xml(reason.message);
+            x = [x, ' line="' num2str(es.line) '"' ...
+                    '>' ...
+                    '<failure message="' ...
+                        get_first_line(reason.message) ...
+                        '">' ...
+                    reason.message ...
+                    '</failure>' ...
+                    '</testcase>' ...
+                    ];
+
+        case 's'
+            % Fill the contents for a skipped test
+            x = [x, '>' ...
+                    '<skipped message="' reason '" />' ...
+                    '</testcase>' ...
+                    ];
+	
+        case '.'
+            % There are no contents for a success, so we just close
+            % the tag
+            x = [x, ' />'];
+	
+        otherwise
+            error('moxunit:notImplemented', ...
+                  'Unrecognised test case result code: %s', lower(c(1)));
+
+    end
+
+
+function string=strip_xml(string)
+% Strip XML tags from a string
+%
+% string=strip_xml(string)
+%
+% Inputs:
+%   string          A string which may contain XML tags Any '<' and '>'
+%                   characters which are not part of an XML tag should
+%                   be escaped as '\<' and '\>', otherwise they will be
+%                   removed.
+%
+% Output:
+%   string          The string with any XML tags removed. Any escaped
+%                   characters ('\<', '\>') remain as they are - still
+%                   escaped.
+%
+% SCL 2015
+
+    % We check to make sure the character before the '<' isn't a '\' or
+    % the start of the string. If it isn't, we capture until the next
+    % non-escaped '>' character and remove the whole chunk. However,
+    % we need to include the leading non-'\' character (the first group)
+    % in the replacement output.
+    string = regexprep(string , '(^|[^\\])(<.*?[^\\]>)', '$1');
```

Comment:
You're right about the '^' as the first character, thanks.
Not sure about an easy fix for `<>` either. It may require some code that just goes over all the characters in a for/while fashion.